File: lib_base/src/main/java/com/android/base/architecture/ui/RefreshView.java
package com.android.base.architecture.ui;

/**
 * <br/>   对下拉刷新的抽象
 * <br/>   Email: 1169654504@qq.com
 *
 * @author Ztiany
 * @version 1.0
 */
public interface RefreshView {

    void autoRefresh();

    void refreshCompleted();

    void setRefreshHandler(RefreshHandler refreshHandler);

    boolean isRefreshing();

    void setRefreshEnable(boolean enable);

    abstract class RefreshHandler {

        public boolean canRefresh() {
            return true;
        }

        public abstract void onRefresh();

    }

}

File: lib_base/src/main/java/com/android/base/architecture/ui/RefreshView.kt
package com.android.base.architecture.ui

/**
 * <br></br>   对下拉刷新的抽象
 * <br></br>   Email: 1169654504@qq.com
 *
 * @author Ztiany
 * @version 1.0
 */
interface RefreshView {

    fun autoRefresh()

    fun refreshCompleted()

    fun setRefreshHandler(refreshHandler: RefreshHandler?)

    val isRefreshing: Boolean

    var isRefreshEnable: Boolean

    abstract class RefreshHandler {

        open fun canRefresh(): Boolean {
            return true
        }

        abstract fun onRefresh()
    }

}